### PR TITLE
Cache integration point result if there are no subscribers and flush when they appear

### PR DESCRIPTION
**Pull request by KovalevAndrey**
## Description

Fixes https://github.com/bumble-tech/appyx/issues/174

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/requestcode/RequestCodeBasedEventStreamImpl.kt [hunk 1]
```diff
@@ -35,23 +42,39 @@ abstract class RequestCodeBasedEventStreamImpl<T : RequestCodeBasedEvent>(
         }
     }
 
+    private fun cacheResultAndFlushOnNewSubscribers(id: Int, flow: MutableSharedFlow<T>, event: T) {
+        cache[id] = event
+        var job: Job? = null
+        job = scope.launch {
+            flow.subscriptionCount.collectLatest { subscriptionCount ->
+                if (subscriptionCount > 0) {
+                    val cachedValue = cache[id]
```
Comment (CherryPerry) [inline libraries/core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/requestcode/RequestCodeBasedEventStreamImpl.kt:51]: Do you need `cache`? Looks like you can capture it in lambda.

**Review by CherryPerry — APPROVED** [review]

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/requestcode/RequestCodeBasedEventStreamImpl.kt [hunk 2]
```diff
@@ -35,23 +41,33 @@ abstract class RequestCodeBasedEventStreamImpl<T : RequestCodeBasedEvent>(
         }
     }
 
+    private fun flushResultWhenSubscribersAppear(flow: MutableSharedFlow<T>, event: T) {
```
Comment (LachlanMcKee) [inline libraries/core/src/main/kotlin/com/bumble/appyx/core/integrationpoint/requestcode/RequestCodeBasedEventStreamImpl.kt:44]: I think in a future PR we should investigate whether we need to remove flows from the events map in cases where a node is never recreated

**Review by LachlanMcKee — APPROVED** [review]

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/node/Node.kt [hunk 3]
```diff
@@ -74,9 +76,12 @@ abstract class Node(
         }
 
     private val lifecycleRegistry = LifecycleRegistry(this)
-
     private var wasBuilt = false
 
+    override val requestCodeClientId: String =
+        buildContext.savedStateMap?.get(RequestCodeClientIdKey) as String?
```
Comment (CherryPerry) [inline libraries/core/src/main/kotlin/com/bumble/appyx/core/node/Node.kt:82]: Let's introduce public `id` property for node.

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/node/Node.kt [hunk 4]
```diff
@@ -211,6 +216,7 @@ abstract class Node(
         plugins<UpNavigationHandler>().any { it.handleUpNavigation() }
 
     companion object {
+        private const val RequestCodeClientIdKey = "node.requestCodeClientId"
```
Comment (CherryPerry) [inline libraries/core/src/main/kotlin/com/bumble/appyx/core/node/Node.kt:219]: FULL_CAPS as in other places? It is not Compose code.

File: libraries/core/src/main/kotlin/com/bumble/appyx/core/node/Node.kt (hunk 3 above)
Comment (CherryPerry) [reply libraries/core/src/main/kotlin/com/bumble/appyx/core/node/Node.kt:82]: https://github.com/bumble-tech/appyx/pull/175/files#diff-a57bab82c13427fb82ff9ac8efdc79f337a988817310bb19a006a9a276df04ddR54

**Review by CherryPerry — APPROVED** [review]

**Pull request merged**